docs/config-generation.md

```diff
@@ -0,0 +1,887 @@
+---
+title: GraphQL Configuration Generation with Tailcall
+description: Migrate REST or Protobuff to GraphQL within minutes
+slug: graphql-configuration-generation-with-tailcall
+sidebar_label: Auto Generation
+---
+
+<head>
+<meta property="og:type" content="article"/>
+  <title>Generate GraphQL Configuration</title>
+</head>
+
+import Tabs from "@theme/Tabs"
+import TabItem from "@theme/TabItem"
+
+# GraphQL Configuration Generation with Tailcall
+
+## What is Configuration Generation?
+
+Configuration generation is the process of automatically generating graphQL configurations from the various sources such as **REST Endpoints**, **Proto files** and already existing **GraphQL configuration** files.
+
+### Problems It Solves
+
+Configuration generation with Tailcall addresses several critical challenges in creating and maintaining GraphQL configurations, ensuring streamlined and error-free processes.
+
+1. **Consistency**: By automatically generating schemas, types etc. Tailcall maintains a consistent structure, making it easier to manage and understand the API.
+2. **Efficiency**:
+   - **Time Savings**: Significantly reduces the time required to generate and update GraphQL configurations by automating the process, allowing developers to focus on core functionalities.
+   - **Quick Migration**: Enables rapid migration from REST or Protbuff to GraphQL, transforming complex API structures into GraphQL schemas within minutes.
+3. **Error Reduction**:
+   - **Minimized Human Errors**: Automates the tedious and error-prone task of manually writing GraphQL configurations, ensuring accuracy and reducing the risk of mistakes.
+   - **Accurate Type Mapping**: Automatically maps API responses to GraphQL types, ensuring correct and precise type definitions without manual intervention.
+
+By addressing these challenges, Tailcall enhances the development workflow, making the process of creating and managing GraphQL configurations more reliable, efficient, and error-free.
+
+### Why is it Hard to Write GraphQL Schemas by Hand?
+
+Writing GraphQL schemas manually presents several challenges that can complicate and slow down the development process:
+
+1. **Complex API Responses**:
+   - **Large and Detailed Responses**: APIs often return extensive and intricate data, making it laborious to map these responses accurately to GraphQL types.
```

Comment:
Someone might have their APIs on Open API or gRPC so why would they need to map those responses?